pkg/paho-mqtt/contrib/riot_iface.c

```diff
@@ -129,13 +132,24 @@ int NetworkConnect(Network *n, char *addr_ip, int port)
         ipv4_addr_from_str((ipv4_addr_t *)&remote.addr, _local_ip)) {
             remote.port = port;
     }
+    else if (IS_USED(MODULE_IPV6_ADDR)) {
+        /* ipvN_addr_from_str modifies input buffer */
```

Comment:
Yes, I think you are right and I also wondered about this but I kept the comment unchanged since I thought it is not directly relevant for my changes and I didn't want to overcomplicate this PR. I will take a look at this again.

Reply:
I can't see a reason not to use `ipvN_addr_from_str()` directly. (And I have also found another place where this is already done: tests/lwip/ip.c:116 and tests/lwip/ip.c:130)

I have removed the comment and the `strncpy`.